packages/editor-ui/src/components/ImportCurlModal.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+	<Modal
+		width="700px"
+		:title="$locale.baseText('importCurlModal.title')"
+		:eventBus="modalBus"
+		:name="IMPORT_CURL"
+		:center="true"
+	>
+		<template slot="content">
+			<div :class="$style.container">
+				<n8n-input-label :label="$locale.baseText('importCurlModal.input.label')">
+					<n8n-input
+						:value="curlCommand"
+						type="textarea"
+						:rows="5"
+						:placeholder="$locale.baseText('importCurlModal.input.placeholder')"
+						@input="onInput"
+						@focus="$event.target.select()"
+						ref="input"
+					/>
+				</n8n-input-label>
+			</div>
+		</template>
+		<template slot="footer">
+			<div :class="$style.modalFooter">
+				<n8n-notice :class="$style.notice" :content="$locale.baseText('ImportCurlModal.notice.content')" />
+				<div>
+					<n8n-button
+						@click="importCurlCommand"
+						float="right"
+						:label="$locale.baseText('importCurlModal.button.label')"
+					/>
+				</div>
+			</div>
+		</template>
+	</Modal>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import Modal from './Modal.vue';
+import { IMPORT_CURL, CURL_INVALID_PROTOCOLS } from '../constants';
+import { showMessage } from './mixins/showMessage';
+import mixins from 'vue-typed-mixins';
+import { INodeUi } from '@/Interface';
+
+export default mixins(showMessage).extend({
+	name: 'ImportCurlModal',
+	components: {
+		Modal,
+	},
+	data() {
+		return {
+			IMPORT_CURL,
+			curlCommand: '',
+			modalBus: new Vue(),
+		};
+	},
+	computed: {
+		node(): INodeUi {
+			return this.$store.getters.activeNode;
+		},
+	},
+	methods: {
+		closeDialog() {
```

Comment:
Nitpick: All typescript methods should have return types (all of those are void but still we should encourage strong typing).